### DISCUSSION TITLE
Build proken by PR #2164 (Add Range Settings to the first statistics page)

### DISCUSSION THREAD

**Discussion by Der-Schubi** [NightscoutFoundation/xDrip#2211] [General]
This PR broke the build: https://github.com/NightscoutFoundation/xDrip/pull/2164
Output:
`> Task :app:compileProdReleaseJavaWithJavac
Gradle may disable incremental compilation as the following annotation processors are not incremental: databinding-compiler-3.4.3.jar (androidx.databinding:databinding-compiler:3.4.3).
Consider setting the experimental feature flag android.enableSeparateAnnotationProcessing=true in the gradle.properties file to run annotation processing in a separate task and make compilation incremental.
/home/cs/Android/Projekte/xDrip/app/build/preprocessor/replace/main/java/com/eveningoutpost/dexdrip/deposit/DepositCallback.java:46: warning: as of release 10, 'var' is a restricted local variable type and cannot be used for type declarations or as the element type of an array
var msg = "Error";
^
warning: vm.showButton.get() is a boxed field but needs to be un-boxed to execute !vm.showButton.get(). This may cause NPE so Data Binding will safely unbox it. You can change the expression and explicitly wrap vm.showButton.get() with safeUnbox() to prevent the warning
warning: vm.showButton.get() is a boxed field but needs to be un-boxed to execute vm.showButton.get() ? prefs.getbool("engineering_mode") : false. This may cause NPE so Data Binding will safely unbox it. You can change the expression and explicitly wrap vm.showButton.get() with safeUnbox() to prevent the warning
vm.showButton.get() is a boxed field but needs to be un-boxed to execute !vm.showButton.get(). This may cause NPE so Data Binding will safely unbox it. You can change the expression and explicitly wrap vm.showButton.get() with safeUnbox() to prevent the warning

vm.showButton.get() is a boxed field but needs to be un-boxed to execute vm.showButton.get() ? prefs.getbool("engineering_mode") : false. This may cause NPE so Data Binding will safely unbox it. You can change the expression and explicitly wrap vm.showButton.get() with safeUnbox() to prevent the warning

/home/cs/Android/Projekte/xDrip/app/build/preprocessor/replace/main/java/com/eveningoutpost/dexdrip/stats/FirstPageFragment.java:94: error: cannot find symbol
TextView rangeView = (TextView) localView.findViewById(R.id.textView_stats_range_set);
^
symbol: variable textView_stats_range_set
location: class id
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
1 error
3 warnings`

**Comment by Navid200** [comment]
@Der-Schubi  Thanks

@tolot27 What could be causing this?
Why would a code fail on some machines and not on the others?  Or, is something else happening here?

**Comment by Navid200** [comment]
@Der-Schubi 
Do you not get the same errors if you compile the release before that (see below)?
https://github.com/NightscoutFoundation/xDrip/releases/tag/2022.07.01

**Comment by Der-Schubi** [comment]
@Navid200 I reverted the 5 commits of the PR and after that it builds without problems.

**Reply by Navid200** [reply]
OK

I don't understand why it compiles for me and others?

I'm not saying there is no error.
I just don't understand why we didn't catch it if it is a real error.

**Comment by Der-Schubi** [comment]
I assume this is a linker problem. A completely clean build works for me.

**Reply by jamorham** [reply]
I was going to say it looks like android studio bug, try `Build -> Clean Project` then do invalidate caches and restart. Make sure your `build.gradle` files are unmodified as well.

**Reply by Der-Schubi** [reply]
Thanks. Invalidate cache it was. :)
